packages/scripts/src/utils/relay.ts

```diff
@@ -0,0 +1,162 @@
+import { BytesLike } from "@ethersproject/bytes";
+import { ArbitrumParentMessenger, GovernorHub, GovernorRootTunnel } from "@uma/contracts-node/typechain/core/ethers";
+import { TransactionDataDecoder } from "@uma/financial-templates-lib";
+import { PopulatedTransaction, Signer } from "ethers";
+const hre = require("hardhat");
+
+export interface ProposedTransaction {
+  to: string;
+  data: string;
+  value: string;
+}
+
+export interface RelayTransaction {
+  to: string;
+  transaction: {
+    name: string;
+    params: {
+      to: string;
+      data?: string;
+      chainId?: string;
+      calls: { to: string; data: string }[];
+    };
+  };
+}
+
+export interface GovernanceMessage {
+  targetAddress: string;
+  tx: PopulatedTransaction;
+}
+
+export type GovernanceMessages = GovernanceMessage[];
+
+export const relayGovernanceRootTunnelMessage = async (
+  targetAddress: string,
+  tx: PopulatedTransaction,
+  governorRootTunnel: GovernorRootTunnel
+): Promise<{
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}> => {
+  if (!tx.data) throw new Error("Transaction has no data");
+  const relayGovernanceData = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  console.log("RelayGovernanceData", relayGovernanceData);
+  const relay = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  const relayMessage = relay.data;
+  if (!relayMessage) throw new Error("Relay message is empty");
+  return { to: governorRootTunnel.address, value: 0, data: relayMessage };
+};
+
+export const relayGovernanceHubMessages = async (
+  messages: GovernanceMessages,
+  governorHub: GovernorHub,
+  chainId: BigNumberish
+): Promise<
+  {
+    to: string;
+    value: BigNumberish;
+    data: BytesLike;
+  }[]
+> => {
+  const calls = messages.map((message) => {
+    if (!message.tx.data) throw new Error("Transaction has no data");
+    return { to: message.targetAddress, data: message.tx.data };
+  });
+  const relayGovernanceData = await governorHub.populateTransaction.relayGovernance(chainId, calls);
+  const relayMessage = relayGovernanceData.data;
+  if (!relayMessage) throw new Error("Relay message is empty");
+  return [{ to: governorHub.address, value: 0, data: relayMessage }];
+};
+
+export const relayGovernanceMessages = async (
+  messages: GovernanceMessages,
+  l1Governor: GovernorHub | GovernorRootTunnel,
+  chainId: number
+): Promise<
+  {
+    to: string;
+    value: BigNumberish;
+    data: BytesLike;
+  }[]
+> => {
+  // The l1 governor for polygon is the GovernorRootTunnel and the l1 governor for the rest of l2's is the GovernorHub
+  const isPolygon = chainId === 137;
+  console.log("isPolygon", isPolygon);
+  if (isPolygon) {
+    const relayedMessages = [];
+    for (const message of messages) {
+      relayedMessages.push(
+        await relayGovernanceRootTunnelMessage(message.targetAddress, message.tx, l1Governor as GovernorRootTunnel)
+      );
```

Comment:
Just OOC, why did you choose to have the relayGovernanceRootTunnelMessage handle one message at once and push the results individually, whereas relayGovernanceHubMessages handles many messages at once?

Not a big deal, was just curious if there was some reason I missed.

Reply:
Good point @mrice32, I decided to be consistent with how the GovernorRootTunnel.[relayGovernance](https://github.com/UMAprotocol/protocol/blob/cee2d0af41cf3a9097f59df43de9f6dc61478edb/packages/core/contracts/polygon-cross-chain-oracle/GovernorRootTunnel.sol#L24)works natively vs the GovernorHub.[relayGovernance](https://github.com/UMAprotocol/protocol/blob/cee2d0af41cf3a9097f59df43de9f6dc61478edb/packages/core/contracts/cross-chain-oracle/GovernorHub.sol#L47). The latter has native support for multiple transactions in one whereas the GovernorRootTunnel's doesn't. I could have used a multicall contract in the destination chain I guess. Do you think it would be better or can you think of a better way to do it?